spring-boot-project/spring-boot-actuator-autoconfigure/src/test/java/org/springframework/boot/actuate/autoconfigure/metrics/export/prometheus/PrometheusMetricsExportAutoConfigurationTests.java

```diff
@@ -128,6 +129,18 @@ public void allowsCustomScrapeEndpointToBeUsed() {
 						.hasSingleBean(PrometheusScrapeEndpoint.class));
 	}
 
```

Comment:
It would be nice if we could test more of the logic in `PushGatewayHandler` somehow. There's logic in the `push` and `shutdown` methods that we don't really have covered. I'm not totally sure how we can do that, perhaps splitting the class so the logic is separate from the scheduling and then testing the logic part with mocks?

Don't worry too much about that if you can't find a way to do it, we can look to it when we merge.

Reply:
Fixed all comments except this one

Reply:
No worries. Thanks for your efforts, we'll try to find a way to add some additional tests when we merge it.

Reply:
Great - I hope it will go into 2.1.0?

Reply:
@philwebb WDYT?

Reply:
ah - I see the added milestones now!